Windows Build Fails Because Of Using Linux Syscall

First of all, thank you for this project; it's a modern and excellent replacement for old crontab implementations on Linux distributions. I use `tasker` in most of my Docker containers, and I'm very happy with it.

----

When building gronx on Windows, there’s a build failure caused by a Linux-specific syscall in [tasker.go:127](https://github.com/adhocore/gronx/blob/89f74f2441582c0e29a81e1ea286b119a336bb98/pkg/tasker/tasker.go#L127) to create a process group ID (PGID).

```
pkg/tasker/tasker.go:127:42: unknown field Setpgid in struct literal of type syscall.SysProcAttr
```

Since this syscall only works on Unix-like systems, the build doesn’t complete successfully on Windows.

```go
exc := exec.Command(sh[0], sh[1], cmd)
exc.Stderr = &buf
exc.SysProcAttr = &syscall.SysProcAttr{Setpgid: true} # Linux Only
```

I’ve found a solution for Windows compatibility and will send a PR to fix this issue.

--- a/pkg/tasker/tasker.go
+++ b/pkg/tasker/tasker.go
@@ -116,38 +116,6 @@ func (t *Tasker) WithContext(ctx context.Context) *Tasker {
 	return t
 }
 
-// Taskify creates TaskFunc out of plain command wrt given options.
-func (t *Tasker) Taskify(cmd string, opt Option) TaskFunc {
-	sh := Shell(opt.Shell)
-
-	return func(ctx context.Context) (int, error) {
-		buf := strings.Builder{}
-		exc := exec.Command(sh[0], sh[1], cmd)
-		exc.Stderr = &buf
-		exc.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
-
-		if t.Log.Writer() != exc.Stderr {
-			exc.Stdout = t.Log.Writer()
-		}
-
-		err := exc.Run()
-		if err == nil {
-			return 0, nil
-		}
-
-		for _, ln := range strings.Split(strings.TrimRight(buf.String(), "\r\n"), "\n") {
-			log.Println(ln)
-		}
-
-		code := 1
-		if exErr, ok := err.(*exec.ExitError); ok {
-			code = exErr.ExitCode()
-		}
-
-		return code, err
-	}
-}
-
 // Shell gives a pair of shell and arg.
 // It returns array of string.
 func Shell(shell ...string) []string {

--- a/pkg/tasker/tasker_other.go
+++ b/pkg/tasker/tasker_other.go
@@ -1,4 +1,5 @@
 //go:build !windows
+// +build !windows
 
 package tasker
 

--- a/pkg/tasker/tasker_other.go
+++ b/pkg/tasker/tasker_other.go
@@ -1,0 +1,43 @@
+//go:build !windows
+
+package tasker
+
+import (
+	"context"
+	"log"
+	"os/exec"
+	"strings"
+	"syscall"
+)
+
+// Taskify creates TaskFunc out of plain command wrt given options.
+func (t *Tasker) Taskify(cmd string, opt Option) TaskFunc {
+	sh := Shell(opt.Shell)
+
+	return func(ctx context.Context) (int, error) {
+		buf := strings.Builder{}
+		exc := exec.Command(sh[0], sh[1], cmd)
+		exc.Stderr = &buf
+		exc.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+		if t.Log.Writer() != exc.Stderr {
+			exc.Stdout = t.Log.Writer()
+		}
+
+		err := exc.Run()
+		if err == nil {
+			return 0, nil
+		}
+
+		for _, ln := range strings.Split(strings.TrimRight(buf.String(), "\r\n"), "\n") {
+			log.Println(ln)
+		}
+
+		code := 1
+		if exErr, ok := err.(*exec.ExitError); ok {
+			code = exErr.ExitCode()
+		}
+
+		return code, err
+	}
+}

--- a/pkg/tasker/tasker_windows.go
+++ b/pkg/tasker/tasker_windows.go
@@ -1,4 +1,5 @@
 //go:build windows
+// +build windows
 
 package tasker
 

--- a/pkg/tasker/tasker_windows.go
+++ b/pkg/tasker/tasker_windows.go
@@ -1,0 +1,46 @@
+//go:build windows
+
+package tasker
+
+import (
+	"context"
+	"log"
+	"os/exec"
+	"strings"
+	"syscall"
+)
+
+// Taskify creates TaskFunc out of plain command wrt given options.
+
+func (t *Tasker) Taskify(cmd string, opt Option) TaskFunc {
+	sh := Shell(opt.Shell)
+
+	return func(ctx context.Context) (int, error) {
+		buf := strings.Builder{}
+		exc := exec.Command(sh[0], sh[1], cmd)
+		exc.Stderr = &buf
+		exc.SysProcAttr = &syscall.SysProcAttr{
+			CreationFlags: syscall.CREATE_NEW_PROCESS_GROUP,
+		}
+
+		if t.Log.Writer() != exc.Stderr {
+			exc.Stdout = t.Log.Writer()
+		}
+
+		err := exc.Run()
+		if err == nil {
+			return 0, nil
+		}
+
+		for _, ln := range strings.Split(strings.TrimRight(buf.String(), "\r\n"), "\n") {
+			log.Println(ln)
+		}
+
+		code := 1
+		if exErr, ok := err.(*exec.ExitError); ok {
+			code = exErr.ExitCode()
+		}
+
+		return code, err
+	}
+}

